src/vmm/src/devices/virtio/balloon/device.rs

```diff
@@ -392,7 +393,10 @@ impl Balloon {
         let mem = self.device_state.mem().unwrap();
         METRICS.stats_updates_count.inc();
 
-        while let Some(head) = self.queues[STATS_INDEX].pop() {
+        while let Some(head) = self.queues[STATS_INDEX]
+            .pop()
+            .map_err(QueueError::InvalidAvailIdx)?
```

Comment:
What if we add a variant in the `BalloonError` like:

```Rust
#[derive(Debug, thiserror::Error, displaydoc::Display)]
pub enum BalloonError {
    ...

    /// Error while processing queue: {0}
    Queue(#[from] QueueError])
}
```

so that we can use `?` and avoid all these `.map_err`?

Reply:
what if the variant was:

```Rust
   InvalidQueue(#[from] InvalidAvailIdx)
```
?

Reply:
that'd work, but then we'd have two different places inside the error enum where the `InvalidAvailIdx` could be stored: inside `QueueError` and inside the new variant. That'd make checking the panic condition more error prone. We could remove it from `QueueError` and give each `<Device>Error` enum an InvalidAvailIdx variant, but I'm not convinced that's actually better than what we have right now :thinking:

Reply:
> . We could remove it from QueueError and give each <Device>Error enum an InvalidAvailIdx variant

That's what I meant before. I think that would make things more readable (avoiding all the `map_err` flood). Am I missing something?